packages/sdk/src/sign.ts

```diff
@@ -498,6 +498,24 @@ export function notificationServiceHash(
         .digest()
 }
 
+export function appRegistryHash(
+    userId: Uint8Array,
+    expiration: bigint, // unix seconds
+    challenge: Uint8Array,
+) {
+    const PREFIX = 'AS_AUTH:'
+    const prefixBytes = new TextEncoder().encode(PREFIX)
+    const expirationBytes = bigIntToBytes(expiration)
+    check(userId.length === 20, 'User ID should be 20 bytes')
+    check(challenge.length === 16, 'Challenge should be 16 bytes')
+    return createHash('sha256')
+        .update(prefixBytes)
+        .update(userId)
+        .update(expirationBytes)
+        .update(challenge)
+        .digest()
+}
+
```

Comment:
After https://github.com/towns-protocol/towns/pull/3292, I'll update this to use web-compatible crypto API